### ISSUE TITLE
Benches generated on external roads using "Build on all paths" option when you have build rights path above it.

### ISSUE THREAD

**Issue by Swift016** [tubbo/openrct2-benchwarmer#56]
Screenshot from Dinky Park, bench on the highway circled in yellow:

<img width="461" alt="Dinky" src="https://user-images.githubusercontent.com/81776470/168743884-21ca32c7-2029-4ffa-9d52-b50771e57fdb.png">

**Comment by RundesBalli** [comment]
I have already noted the "outside of the park" error in this issue:
https://github.com/tubbo/openrct2-benchwarmer/issues/25

Maybe it is because the path is elevated and not on the ground as mentioned by @pencil here:
https://github.com/tubbo/openrct2-benchwarmer/issues/25#issuecomment-699258719

Do you have construction rights in this area?

**Comment by tubbo** [comment]
Yeah, I noticed this too...there's no way to know whether what you're placing a bench/bin on is a "road" because roads are just asphalt paths with a dotted line in the middle. As far as the game is concerned, they are all paths. Also, you could theoretically build a road inside your park that guests can walk on, and you would want benches on either side of that path, so I can't see a good way of fixing this issue.

**Comment by Swift016** [comment]
@tubbo Your logic is pretty faulty.

1) Anyone who is drawing "roads" on their map as decoration is never going to use the "build all" option for the exact reason you specified, and if they do, they should expect the results they get, making your objection a moot point.

2) In this specific bug, the "road" is quite literally outside the purview of the player's park. It's a no-brainer that the fix is to NOT place benches on paths that are not within the build rights of the player. I'm confused why I need to even explain that.

**Comment by pencil** [comment]
This is not a bug, it's working as expected: If you have construction rights in an area, which you do you have here, then you are allowed to place additions on paths that are elevated. You can manually place additions too in Dinky Park on the "road" (which is really just a path, ass @tubbo pointed out) once you obtain construction rights.

![path](https://user-images.githubusercontent.com/812086/196263897-3a47f81a-2c07-493a-b608-32505c47250c.png)

**Comment by Swift016** [comment]
@pencil This adds important clarification. Thank you. I had no idea you could buy build rights and quite literally delete the pre-placed roads if they are elevated. This definitely seems like an oversight in the base game.

<img width="491" alt="dinky" src="https://user-images.githubusercontent.com/81776470/196270265-b1cdba40-2c09-4f6f-8a17-76ed6b184ed7.png">

In that case, one could hypothetically scan the park on map creation for paths that fall into this very niche category and remove them from consideration by the "build on all" function, but then you'd have to keep track of if those paths were later removed/replaced by hand by the player (and presumably wanting them to have bins/benches).

**Comment by pencil** [comment]
Yeah, definitely a bit of a quirk in the game. In theory it might be possible to check if the footpath surface type "road" (which is what causes those white lines between the paths) is present in the tile but I haven't check if it's exposed via the plugin scripting API.

**Comment by pencil** [comment]
Oh, I should have read the whole thread first. As @tubbo pointed out, there might be situations where you actually want benches on a manually built path with the road surface. This could potentially be solved with a setting but I don't know if it's worth the effort.

> one could hypothetically scan the park on map creation

This would only work when you start a new scenario. If you load a saved park, there would be no way to determine which roads were there to begin with and which ones were built by the player.

**Comment by tubbo** [comment]
> In this specific bug, the "road" is quite literally outside the purview of the player's park. It's a no-brainer that the fix is to NOT place benches on paths that are not within the build rights of the player. I'm confused why I need to even explain that.

@Swift016 Apologies, when I read the issue, I thought the ask/bug was specific to roads, but in fact this would apply to any path outside the park gates. This is why I mentioned that I wasn't sure how to fix this and added the "help wanted" label, in lieu of just making a pull request.

Since we're using the in-game actions API, the plugin essentially uses the same logic as when you manually place objects. This not only charges you appropriately, but also validates whether you're allowed to place the thing in the first place. So if the plugin is able to place a bench/bin, you should be able to place things there manually as well.

It may be that the confusion here stemmed from the construction rights aspect of the game, which I can attest to as well. I wonder if it's possible to know whether we're building on a tile you actually own vs one that you have construction rights on. If so, maybe we can have a toggle that will prevent placing benches/bins on construction rights tiles? It feels like that might be a sane default as well, because you don't get a whole lot of benefit from placing stuff on paths outside your park gates.

**Comment by pencil** [comment]
>  I wonder if it's possible to know whether we're building on a tile you actually own vs one that you have construction rights on.

Yes, the code (that duplicates the game logic, as you said) is here: https://github.com/tubbo/openrct2-benchwarmer/blob/cf6c9c497ea0ce04574e8847c52afbb42e3cc2e3/src/add.ts#L120-L123
